crates/editor/src/clangd_ext.rs

```diff
@@ -0,0 +1,119 @@
+use std::path::PathBuf;
+
+use anyhow::Context as _;
+use gpui::{View, ViewContext, WindowContext};
+use language::Language;
+use url::Url;
+
+use crate::{element::register_action, Editor, SwitchSourceHeader};
+
+pub fn apply_related_actions(editor: &View<Editor>, cx: &mut WindowContext) {
+    let is_c_related = editor.update(cx, |editor, cx| {
+        editor
+            .buffer()
+            .read(cx)
+            .all_buffers()
+            .iter()
+            .any(|b| match b.read(cx).language() {
+                Some(l) => is_c_language(l),
```

Comment:
I wonder if we could do a deeper check here (the one in rust_analyzer_ext.rs is sort of sloppy/wrong and relied on a fact that there's a single Rust langserver, whereas for C or C++ that could be much more different).

We can call `language_server_ids_for_buffer` here similar to what is done below (maybe, we should extract that whole thing into a separate function to call twice?) and ensure that the ext action is registered from proper LSP only.

Feel free to do the same for rust-analyzer if you want to.